native-platform/src/main/cpp/posix.cpp

```diff
@@ -52,10 +53,38 @@ Java_net_rubygrapefruit_platform_internal_jni_NativeLibraryFunctions_getSystemIn
     env->SetObjectField(info, osNameField, char_to_java(env, machine_info.sysname, result));
     jfieldID osVersionField = env->GetFieldID(infoClass, "osVersion", "Ljava/lang/String;");
     env->SetObjectField(info, osVersionField, char_to_java(env, machine_info.release, result));
-    jfieldID machineArchitectureField = env->GetFieldID(infoClass, "machineArchitecture", "Ljava/lang/String;");
-    env->SetObjectField(info, machineArchitectureField, char_to_java(env, machine_info.machine, result));
     jfieldID hostnameField = env->GetFieldID(infoClass, "hostname", "Ljava/lang/String;");
     env->SetObjectField(info, hostnameField, char_to_java(env, machine_info.nodename, result));
+
+    jfieldID machineArchitectureField = env->GetFieldID(infoClass, "machineArchitecture", "Ljava/lang/String;");
+#ifndef __APPLE__
+    env->SetObjectField(info, machineArchitectureField, char_to_java(env, machine_info.machine, result));
+#else
+    // On macOS, uname() reports the architecture of the current binary.
+    // Instead, use a macOS specific sysctl() to query the CPU name, which can be mapped to the architecture
+    int mib[5];
```

Comment:
why 5? I'm having trouble finding official documentation for "machdep.cpu.brand_string"

Reply:
It's 5 because that's what it was in the example I copied it from. 4 also works. 3 and lower do not.

It's kind of documented on the sysctl man page.